src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -86,67 +104,660 @@ static DamageTable D_80BAA468 = {
+        this->unk_410 = func_8013BB34(globalCtx, sp23, D_80BAA4A8[arg2->unk0]);
+    }
+
+    if (this->unk_410 == NULL) {
+        return false;
+    }
+
+    if ((this->unk_434 != 0) && (this->unk_436 >= 0)) {
+        temp = sp26;
+    } else {
+        temp = arg2->unk4;
+    }
+
+    if (arg2->unk8 < temp) {
+        this->unk_424 = (temp - arg2->unk8) + 0xFFFF;
+    } else {
+        this->unk_424 = arg2->unk8 - temp;
+    }
+
+    this->unk_430 = sp26 - temp;
+    temp = this->unk_410->count - 2;
+    this->unk_428 = this->unk_424 / temp;
+    this->unk_438 = 0;
+    this->unk_42C = (this->unk_430 / this->unk_428) + 2;
+    this->unk_43C = 0;
+    return true;
+}
+
+s32 func_80BA9110(EnBaba* this, GlobalContext* globalCtx, struct_80133038_arg2* arg2) {
+    s32 ret;
+
+    switch (arg2->unk0) {
+        default:
+            ret = false;
+            break;
+
+        case 2:
+            ret = func_80BA8F88(this, globalCtx, arg2);
+            break;
+
+        case 1:
+            ret = true;
+            break;
+    }
+    return ret;
+}
+
+s32 func_80BA9160(EnBaba* this, GlobalContext* globalCtx) {
+    f32 sp7C[265];
+    Vec3f sp70;
+    Vec3f sp64;
+    Vec3f sp58;
+    s32 sp54 = 0;
+    s32 sp50 = 0;
+    s32 pad;
+
+    func_8013AF00(sp7C, 3, this->unk_410->count + 3);
+
+    if (this->unk_438 == 0) {
+        sp58 = gZeroVec3f;
+
+        func_8013B6B0(this->unk_410, &this->unk_420, &this->unk_430, this->unk_428, this->unk_424, &this->unk_42C, sp7C,
+                      &sp58, this->unk_436);
+        func_8013B878(globalCtx, this->unk_410, this->unk_42C, &sp58);
+        this->actor.world.pos.y = sp58.y;
+        this->unk_438 = 1;
+    } else {
+        sp58 = this->unk_414;
+    }
+
+    this->actor.world.pos.x = sp58.x;
+    this->actor.world.pos.z = sp58.z;
+
+    if (func_8013AD6C(globalCtx)) {
+        sp54 = this->unk_430;
+        sp50 = this->unk_42C;
+        sp58 = this->actor.world.pos;
+    }
+
+    this->unk_414 = gZeroVec3f;
+
+    if (func_8013B6B0(this->unk_410, &this->unk_420, &this->unk_430, this->unk_428, this->unk_424, &this->unk_42C, sp7C,
+                      &this->unk_414, this->unk_436)) {
+        this->unk_43C = 1;
+    } else {
+        sp70 = this->actor.world.pos;
+        sp64 = this->unk_414;
+        this->actor.world.rot.y = Math_Vec3f_Yaw(&sp70, &sp64);
+    }
+
+    if (func_8013AD6C(globalCtx)) {
+        this->unk_430 = sp54;
+        this->unk_42C = sp50;
+        this->unk_414 = sp58;
+    }
+
+    return false;
+}
+
+void func_80BA93AC(EnBaba* this, GlobalContext* globalCtx) {
+    if (this->unk_434 != 1) {
+        if (this->unk_434 == 2) {
+            gSaveContext.weekEventReg[58] |= 0x40;
+            this->unk_40A |= 2;
+            func_80BA9160(this, globalCtx);
+        }
+    } else {
+        this->unk_40C = 3;
+        this->unk_1E0 = 10800;
+        this->actor.speedXZ = 0.0f;
+        Enemy_StartFinishingBlow(globalCtx, &this->actor);
```

Comment:
```suggestion
        Enemy_StartFinishingBlow(globalCtx, &this->actor);
```
wat

we can KILL GRANDMA?